frontend/src/components/SortingHat.tsx

```diff
@@ -1,14 +1,97 @@
+
+interface House {
+    name: string;
+    score: number
+}
 
 export default function SortingHat() {
+    const [housesScore, setHousesScore] = useState<Array<House>>([{name: 'Gryffindor', score: 0}, {
+        name: 'Ravenclaw',
+        score: 0
+    }, {name: 'Hufflepuff', score: 0}, {name: 'Slytherin', score: 0}]);
+    let text;
+    useEffect(() => {
+        givingScore(Attribute1.BRAVE);
+        givingScore(Attribute2.FAITHFUL);
+        givingScore(Attribute3.REBELLIOUS);
+    }, []);
+
+    const givingScore = (attribute: string) => {
+
+        if (attribute === Attribute1.BRAVE || attribute === Attribute2.COURAGEOUS || attribute === Attribute3.DETERMINED) {
+            setHousesScore(prevState => {
+                const newState = prevState.map(house => {
+                    if (house.name === 'Gryffindor') {
+                        return {...house, score: house.score + 1};
+                    }
+                    return house;
+                });
+
+                newState.sort((a: House, b: House) => b.score - a.score);
+                return newState;
+            });
+        }
+
+        if (attribute === Attribute1.JUST || attribute === Attribute2.FAITHFUL || attribute === Attribute3.DILIGENT) {
+            setHousesScore(prevState => {
+                const newState = prevState.map(house => {
+                    if (house.name === 'Hufflepuff') {
+                        return {...house, score: house.score + 1};
+                    }
+                    return house;
+                });
+                newState.sort((a: House, b: House) => b.score - a.score);
+
+                return newState;
+            });
+        }
+
+        if (attribute === Attribute1.CLEVER || attribute === Attribute2.CREATIVE || attribute === Attribute3.INTROVERT) {
+            setHousesScore(prevState => {
+                const newState = prevState.map(house => {
+                    if (house.name === 'Ravenclaw') {
+                        return {...house, score: house.score + 1};
+                    }
+                    return house;
+                });
+                newState.sort((a: House, b: House) => b.score - a.score);
+
+                return newState;
+            });
+        }
+
+        if (attribute === Attribute1.AMBITIOUS || attribute === Attribute2.ASSERTIVE || attribute === Attribute3.REBELLIOUS) {
+            setHousesScore(prevState => {
+                const newState = prevState.map(house => {
+                    if (house.name === 'Slytherin') {
+                        return {...house, score: house.score + 1};
+                    }
+                    return house;
+                });
+                newState.sort((a: House, b: House) => b.score - a.score);
+
+                return newState;
+            });
+        }
+    }
+
+    if (housesScore[0].score > housesScore[1].score) {
+        text = <p className="SpeakingHat">Das ist ganz klar, du bist ein {housesScore[0].name}!</p>;
```

Comment:
Isn't this rather .. `speakingHatsDecision` (whatever you can discuss this futher)